sdk/storage/azure-storage-queue/src/test/java/com/azure/storage/queue/spock/QueueServiceAPITests.groovy

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.queue.spock
+
+import com.azure.storage.queue.models.*
+import org.junit.Assert
+import spock.lang.Unroll
+
+class QueueServiceAPITests extends APISpec {
+
+    def setup() {
+        primaryQueueServiceClient = queueServiceBuilderHelper(interceptorManager).buildClient()
+    }
+    
+    def "Get queue client from queue service client"() {
+        when:
+        def queueClient = primaryQueueServiceClient.getQueueClient(testResourceName.randomName("queue", 16))
+        queueClient.enqueueMessage("Expecting an exception")
+        then:
+        def e = thrown(StorageErrorException)
+        QueueTestHelper.assertExceptionStatusCodeAndMessage(e, 404, StorageErrorCode.QUEUE_NOT_FOUND)
+    }
+
+    def "Create queue from queue service client"() {
+        when:
+        def queueClientResponse = primaryQueueServiceClient.createQueue(testResourceName.randomName("queue", 16))
+        def enqueueMessageResponse = queueClientResponse.value().enqueueMessage("Testing service client creating a queue")
+        then:
+        QueueTestHelper.assertResponseStatusCode(queueClientResponse, 201)
+        QueueTestHelper.assertResponseStatusCode(enqueueMessageResponse, 201)
+    }
+
+    def "Create queues with duplicate name from queue service client"() {
+        given:
+        def queueName = testResourceName.randomName("queue", 16)
+        when:
+        primaryQueueServiceClient.createQueue(queueName)
+        def queueClientResponse = primaryQueueServiceClient.createQueue(queueName)
+        then:
+        QueueTestHelper.assertResponseStatusCode(queueClientResponse, 204)
+    }
+
+    @Unroll
+    def "Create queue with invalid name from queue service client"() {
+        when:
+        primaryQueueServiceClient.createQueue(queueName)
+        then:
+        def e = thrown(StorageErrorException)
+        QueueTestHelper.assertExceptionStatusCodeAndMessage(e, statusCode, errMesage)
+        where:
+        queueName      | statusCode | errMesage
+        "a_b"          | 400        | StorageErrorCode.INVALID_RESOURCE_NAME
+        "-ab"          | 400        | StorageErrorCode.INVALID_RESOURCE_NAME
+        "a--b"         | 400        | StorageErrorCode.INVALID_RESOURCE_NAME
+        // null | 400 | "InvalidResourceName" TODO: Need to fix the RestProxy before having null parameter
```

Comment:
We should probably be asserting not null in the builder, no?

Reply:
I have made changes about checking null for the resource name